plugins/verifier/licensechecker/licensechecker.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/deislabs/ratify/plugins/verifier/licensechecker/utils"
+
+	"github.com/deislabs/ratify/pkg/common"
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore"
+	_ "github.com/deislabs/ratify/pkg/referrerstore/oras"
+	"github.com/deislabs/ratify/pkg/verifier"
+	"github.com/deislabs/ratify/pkg/verifier/plugin/skel"
+)
+
+type PluginConfig struct {
+	Name            string   `json:"name"`
+	AllowedLicenses []string `json:"allowedLicenses"`
+}
+
+type PluginInputConfig struct {
+	Config PluginConfig `json:"config"`
+}
+
+func main() {
+	skel.PluginMain("licensechecker", "1.0.0", VerifyReference, []string{"1.0.0"})
+}
+
+func parseInput(stdin []byte) (*PluginConfig, error) {
+	conf := PluginInputConfig{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse stdin for input: %v", err)
+	}
+
+	return &conf.Config, nil
+}
+
+func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, descriptor ocispecs.ReferenceDescriptor, store referrerstore.ReferrerStore) (*verifier.VerifierResult, error) {
+	input, err := parseInput(args.StdinData)
+	if err != nil {
+		return nil, err
+	}
+
+	allowedLicenses := utils.LoadAllowedLicenses(input.AllowedLicenses)
+
+	ctx := context.Background()
+	referenceManifest, err := store.GetReferenceManifest(ctx, subjectReference, descriptor)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, blobDesc := range referenceManifest.Blobs {
+		refBlob, err := store.GetBlobContent(ctx, subjectReference, blobDesc.Digest)
+		if err != nil {
+			return nil, err
+		}
+
+		spdxDoc, err := utils.BlobToSPDX(refBlob)
+		if err != nil {
+			return nil, err
+		}
+
+		packageLicenses := utils.GetPackageLicenses(*spdxDoc)
+		disallowedLicenses := utils.FilterPackageLicenses(packageLicenses, allowedLicenses)
+
+		if len(disallowedLicenses) > 0 {
```

Comment:
Should we default to success or failure if there are no package licenses present but there are allowed licenses specified?

Reply:
My personal opinion is the default should be to succeed with a warning. I will have to look at what mechanisms we currently have to make that possible. If an external package doesn't have a declared license it could be a long time to remedy that in a way that would be picked up by this check.

There are a couple license-specific settings I think it would be beneficial to expose to the user to add flexibility:
- What to do if there isn't a declared license for a package
- What to do if the declared license doesn't match the concluded license
- What to do for combinations of licenses (SPDX supports OR and AND for license declarations)

I am currently looking at including a rego executor so that the logic for these checks can be made more concise and to make it easier to take action on other SPDX fields.

The aim of this PR is to provide a starting point with practical use of a real SBoM.